pkgs/top-level/static.nix

```diff
@@ -77,35 +77,251 @@ self: super: let
       });
     };
 
-in {
+in rec {
   stdenv = foldl (flip id) super.stdenv staticAdapters;
 
+  at-spi2-atk = super.at-spi2-atk.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
+    ];
+    NIX_CFLAGS_COMPILE = "-DG_LOG_DOMAIN=\"\"\"\"";
+  });
+
+  at-spi2-core = super.at-spi2-core.overrideAttrs (old: {
+    NIX_MESON_DEPENDENCY_STATIC = true;
+    nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+      super.mesonShlibsToStaticHook
```

Comment:
maybe this hook should be propagated by meson when `targetPlatform.isStatic` ?

Reply:
Would that work ? 

<details>

```patch
diff --git a/pkgs/development/tools/build-managers/meson/default.nix b/pkgs/development/tools/build-managers/meson/default.nix
index 60ab7605944..219160c66a5 100644
--- a/pkgs/development/tools/build-managers/meson/default.nix
+++ b/pkgs/development/tools/build-managers/meson/default.nix
@@ -1,13 +1,14 @@
-{ lib
+{ stdenv
+, lib
 , python3
-
+, mesonShlibsToStaticHook
 , writeTextDir
 , substituteAll
 , fetchpatch
 , installShellFiles
 }:
 
-python3.pkgs.buildPythonApplication rec {
+python3.pkgs.buildPythonApplication (rec {
   pname = "meson";
   version = "0.57.1";
 
@@ -98,4 +99,6 @@ python3.pkgs.buildPythonApplication rec {
     maintainers = with maintainers; [ jtojnar mbe ];
     platforms = platforms.all;
   };
-}
+} // lib.optionalAttrs stdenv.hostPlatform.isStatic {
+  propagatedBuildInputs = [ mesonShlibsToStaticHook ];
+})
```

</details>

Reply:
actually putting this in propagatedBuildInputs is wrong because it also affects transitive dependencies. What about incorporating this to pkgs/development/tools/build-managers/meson/setup-hook.sh gated behind `if [ -n $mesonForceReplaceSharedLibsByStaticLibs ]` and `$mesonForceReplaceSharedLibsByStaticLibs` defaults to `targetPlatform.isStatic`  (you can pass the value from nix to setup-hook.sh by using the fact that according to the doc of setupHook https://nixos.org/manual/nixpkgs/stable/#ssec-fixup-phase @envvar@ is replaced by the value of envvar during the build of meson)